src/auth/ThreeStepOwned.sol

```diff
@@ -42,30 +43,37 @@ abstract contract ThreeStepOwned {
                              OWNERSHIP LOGIC
     //////////////////////////////////////////////////////////////*/
 
-    function renounceOwnership() public virtual onlyOwner {
-        owner = address(0);
-
-        emit OwnerUpdated(msg.sender, address(0));
-    }
-
-    function initOwnershipUpdate(address _newOwnerCandidate) public virtual onlyOwner {
-        _ownerCandidate = _newOwnerCandidate;
+    function setOwner(address newOwner) public virtual onlyOwner {
+        _ownerCandidate = newOwner;
 
-        emit OwnerUpdateInitiated(msg.sender, _newOwnerCandidate);
+        emit OwnerUpdateInitiated(msg.sender, newOwner);
     }
 
-    function confirmOwnershipUpdate() public virtual {
-        if (_updateConfirmedByCandidate) {
+    function confirmOwner() public virtual {
+        if (_ownerCandidateConfirmed) {
             require(msg.sender == owner, "UNAUTHORIZED");
-
-            _updateConfirmedByCandidate = false;
-            owner = _ownerCandidate;
-
-            emit OwnerUpdated(msg.sender, owner);
+            
+            delete _ownerCandidateConfirmed;
+            
+            address newOwner = _ownerCandidate;
+            
+            delete _ownerCandidate;
```

Comment:
Same here, worth to do this?

Reply:
In this case, imo, it's not needed neither. The `_ownerCandidate` will stay matching the `owner`, so nothing bad could happen, just re-confirming the same current owner. I think doesn't worth to delete it.

Reply:
hm, nope, you're right that there is no risk of rug, and I guess removing this delete keeps slot warm?

Reply:
Might worth to do some gas snapshot test to verify. But I'm guessing is cheaper to avoid the delete, yeah. That's why I've chosen to avoid cleaning the storage.

Reply:
good stuff